src/mame/drivers/mpu4.cpp

```diff
@@ -3071,3 +3021,258 @@ void mpu4_state::mpu4crys(machine_config &config)
 	upd.add_route(ALL_OUTPUTS, "lspeaker", 1.0);
 	upd.add_route(ALL_OUTPUTS, "rspeaker", 1.0);
 }
+
+void mpu4_state::init_m4debug()
+{
+	// many original barcrest / bwb sets have identification info around here
+	// this helps with sorting
+	uint8_t *src = memregion( "maincpu" )->base();
+	int size = memregion( "maincpu" )->bytes();
+
+	// m4richfm__e only has 0x004000
+	if (size < 0x10000)
+		return;
+
+	for (int j=0;j<size;j+=0x10000)
+	{
+		if (size>0x10000) printf("\nblock 0x%06x:\n",j);
+		printf("\ncopyright string:\n");
+		for (int i = 0xffe0; i<0xfff0; i++)
+		{
+			printf("%c", src[j+i]);
+		}
+		printf("\n\nidentification string:\n");
+		for (int i = 0xff28; i<0xff30; i++)
+		{
+			printf("%c", src[j+i]);
+		}
+		printf("\n");
+
+		printf("flags ENABLED\n");
+		for (int i = 0xff30; i<0xff9f; i+=2)
+		{
+			uint8_t enable = src[j+i+0];
+			uint8_t letter = src[j+i+1];
+
+			if (enable == 0x00)
+			{
+				printf("%c", letter);
+			}
+		}
+		printf("\n");
+		printf("flags DISABLED\n");
+		for (int i = 0xff30; i<0xff9f; i+=2)
+		{
+			uint8_t enable = src[j+i+0];
+			uint8_t letter = src[j+i+1];
+
+			if (enable == 0xff)
+			{
+				if (letter != 0xff)
+					printf("%c", letter);
+			}
+		}
+		printf("\n");
+
+
+	}
+}
+
+void mpu4_state::mod4oki_chr(machine_config &config)
+{
+	mod4oki(config);
+	MPU4_CHARACTERISER_PAL(config, m_characteriser, 0);
+}
+
+void mpu4_state::mod4oki_cheatchr(machine_config &config)
+{
+	mod4oki(config);
+	MPU4_CHARACTERISER_PAL(config, m_characteriser, 0);
+	m_characteriser->set_cpu_tag("maincpu");
+	m_characteriser->set_allow_6809_cheat(true);
+//	m_characteriser->set_lamp_table(xxx_lamp_scramble);
+}
+
+void mpu4_state::mod4oki_cheatchr_tri98(machine_config &config)
+{
+	mod4oki(config);
+	MPU4_CHARACTERISER_PAL(config, m_characteriser, 0);
+	m_characteriser->set_cpu_tag("maincpu");
+	m_characteriser->set_allow_6809_cheat(true);
+	m_characteriser->set_lamp_table(mpu4_characteriser_pal::m4dtri98_lamp_scramble);
+}
+
+void mpu4_state::mod4oki_cheatchr_lv(machine_config &config)
+{
+	mod4oki(config);
+	MPU4_CHARACTERISER_PAL(config, m_characteriser, 0);
+	m_characteriser->set_cpu_tag("maincpu");
+	m_characteriser->set_allow_6809_cheat(true);
+	m_characteriser->set_lamp_table(mpu4_characteriser_pal::m4lv_lamp_scramble);
+}
+
+void mpu4_state::mod4oki_cheatchr_m683(machine_config &config)
+{
+	mod4oki(config);
+	MPU4_CHARACTERISER_PAL(config, m_characteriser, 0);
+	m_characteriser->set_cpu_tag("maincpu");
+	m_characteriser->set_allow_6809_cheat(true);
+	m_characteriser->set_lamp_table(mpu4_characteriser_pal::m683_lamp_scramble);
+}
```

Comment:
You could make a helper member function that takes the `machine_config` and the scramble table as parameters to cut these function bodies down from five lines to one.  It should be a net saving of lines.